### ISSUE TITLE
Clicking Inside a selected text does not clear the selection

### ISSUE THREAD

**Issue by icuurd12b42** [Microsoft/vscode#34559]
- VSCode Version: Code 1.16.1 (27492b6bf3acb0775d82d2f87b25a93490673c6d, 2017-09-14T16:38:23.027Z)
- OS Version: Windows_NT x64 10.0.15063
- Extensions: 

Extension|Author (truncated)|Version
---|---|---
vscode-autohotkey|sle|0.2.1

Clicking Inside a selected text does not clear the selection. An annoying bug, or useless feature... If you have a bunch of lines selected and you are done with it and want to continue working, you cannot simply click anywhere to clear the selection.  You have to click on a non selected line, clicking on the right of the selected text, in the whitespace simply wont do.

Long story short, the text should deselect no matter where I click, if I click inside the selected text, to perform a drag for example (which no one in his right mind would ever do anyway), the text should deselect upon left mouse release, because guess what, I'm done with this action! If you want to keep the selected text after a drag, then at least de-select if the click was not a drag

---

Steps to Reproduce:

1.Select a number of lines
2.Click the white space on the right or inside the selection. the text does not de-select

Reproduces without extensions: Yes/No

**Comment by jhasse** [comment]
As a workaround you can set `"editor.dragAndDrop": false` in your settings.

**Comment by icuurd12b42** [comment]
Thanks. Still the default, Still the implementation needs work
